EventMixing/src/MixBackgroundFrames_module.cc

```diff
@@ -121,28 +137,58 @@ namespace mu2e {
   MixBackgroundFramesDetail::MixBackgroundFramesDetail(const Parameters& pars, art::MixHelper& helper)
     : spm_{ pars().mu2e().products(), helper }
     , pbiTag_{ pars().mu2e().protonBunchIntensityTag() }
-    , meanEventsPerProton_{ pars().mu2e().meanEventsPerProton() }
     , debugLevel_{ pars().mu2e().debugLevel() }
     , engine_{helper.createEngine(art::ServiceHandle<SeedService>()->getSeed())}
     , urbg_{ engine_ }
     , totalBkgCount_(0)
     , skipFactor_{ pars().mu2e().skipFactor() }
     , writeEventIDs_{ pars().mu2e().writeEventIDs() }
+    , simStageEfficiencyTags_{ pars().mu2e().simStageEfficiencyTags() }
+    , meanEventsPerPOTFactors_{ pars().mu2e().meanEventsPerPOTFactors() }
+    , mixingMeanOverride_(false)
   {
     if(writeEventIDs_) {
       helper.produces<art::EventIDSequence>();
     }
+    if (pars().mu2e().meanEventsPerProton(meanEventsPerProton_)) {
+      mixingMeanOverride_ = true;
+    }
   }
 
   //================================================================
   void MixBackgroundFramesDetail::startEvent(const art::Event& event) {
     pbi_ = *event.getValidHandle<ProtonBunchIntensity>(pbiTag_);
     if(debugLevel_ > 0)std::cout << " Starting event mixing, Intensity = " << pbi_.intensity() << std::endl;
+
+    SimBookkeeper const& simbookkeeper = _simbookkeeperH.get(event.id());
```

Comment:
Talking to the DB should be conditional.  If one specified meanEventsPerProton no handle should be created here.

Reply:
Done (commit c6d970e47)